### PR TITLE
resume upload part concurrently

### PR DESCRIPTION
**Pull request by lihsai0**
KODO-18759

### CODE REVIEW

File: qiniu/services/storage/uploaders/form_uploader.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+        self,
+        up_token,
+        **kwargs
+    ):
+        """
+        Parameters
+        ----------
+        up_token: str
+        kwargs
+            key, file_crc32, costume_vars, metadata
```
Comment (bachue) [inline qiniu/services/storage/uploaders/form_uploader.py:183]: @lihsai0 `costume` => `custom`

Comment (bachue) [reply qiniu/services/storage/uploaders/form_uploader.py:183]: @lihsai0 `file_crc32` => `crc32`

File: qiniu/services/storage/uploaders/form_uploader.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+        data: IOBase
+        data_size: int
+        modify_time: int
+        part_size: int
+        mime_type: str
+        metadata: dict
+        file_name: str
+        costume_vars: dict
+        kwargs
+            up_token, file_crc32
```
Comment (bachue) [inline qiniu/services/storage/uploaders/form_uploader.py:57]: @lihsai0 `file_crc32` => `crc32`

File: qiniu/services/storage/uploaders/form_uploader.py [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+        key: str
+        file_path: str
+        data: IOBase
+        data_size: int
+        modify_time: int
+        part_size: int
+        mime_type: str
+        metadata: dict
+        file_name: str
+        costume_vars: dict
```
Comment (bachue) [inline qiniu/services/storage/uploaders/form_uploader.py:55]: @lihsai0 `costume` => `custom`

File: qiniu/services/storage/uploaders/resume_uploader_v2.py [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+        file_name: str
+        key: str
+        context: _ResumeUploadV2Context
+
+        Returns
+        -------
+        _ResumeUploadV2Context
+        """
+        if not isinstance(context, _ResumeUploadV2Context):
+            raise TypeError('context must be an instance of _ResumeUploadV1Context')
```
Comment (bachue) [inline qiniu/services/storage/uploaders/resume_uploader_v2.py:38]: @lihsai0 `_ResumeUploadV1Context` => `_ResumeUploadV2Context`

File: qiniu/compat.py [hunk 5]
```diff
@@ -51,6 +51,13 @@ def s(data):
     def u(data):
         return unicode(data, 'unicode_escape')  # noqa
 
+    def is_seekable(data):
+        try:
+            data.seek(0, 1)
```
Comment (bachue) [inline qiniu/compat.py:56]: @lihsai0 `1` => `os.SEEK_CUR`

**Review by bachue — APPROVED** [review]

File: qiniu/services/storage/uploaders/form_uploader.py (hunk 2 above)
Comment (lihsai0) [reply qiniu/services/storage/uploaders/form_uploader.py]: `file_crc32` => `crc32_int`

**Review by bachue — APPROVED** [review]

**Pull request merged**